### PR TITLE
Phase control of Tektronix AFG3152C

### PR DESCRIPTION
**Pull request by FlyingCurryMonster**
I was using the AFG3152C and noticed I couldn't control the phase of the signal with pymeasure, so I added in Instrument.control method for that.

### CODE REVIEW

File: pymeasure/instruments/tektronix/afg3152c.py [hunk 1]
```diff
@@ -114,6 +120,22 @@ class Channel:
...
+        validator = strict_range,
+        values = PHASE_LIMIT['RAD']
+    )
+    
+    phase_deg = Instrument.control(
+        "phase:adjust?", "phase:adjust %e DEG",
+        """ A floating point property that controls the phase in radian units.
+        This property can be set.""",
+        validator=strict_range,
+        values = PHASE_LIMIT['DEG']
```
Comment (LongnoseRob) [inline pymeasure/instruments/tektronix/afg3152c.py:137]: Probably a simple c&p mistake, for this property the text should be changed to `"...  in degrees."`

**Review by LongnoseRob — COMMENTED** [review]
Thank you for the addition of the phase-control properties!

I found a small mistake in the doc-string for the `phase_deg` property.

File: pymeasure/instruments/tektronix/afg3152c.py (hunk 1 above)
Comment (FlyingCurryMonster) [reply pymeasure/instruments/tektronix/afg3152c.py:137]: Oops yea it was a copy and paste mistake.  And the string is actually different for phase_rad and phase_deg.  phase_deg ends with %e DEG, while phase_rad ends with %e RAD.

Comment (bilderbuchi) [reply pymeasure/instruments/tektronix/afg3152c.py]: no, the _query_ string, meaning the `get_command` argument is the same -- what's up with that? what happens if you do 
```
print(instr.phase_deg)
print(instr.phase_rad)
```
?

**Comment by bilderbuchi** [comment]
The linter found some formatting issues (check in the changed files view, or run flake8 locally)

File: pymeasure/instruments/tektronix/afg3152c.py (hunk 1 above)
Comment (FlyingCurryMonster) [reply pymeasure/instruments/tektronix/afg3152c.py]: It looks like phase_deg won't return the phase in degrees unless it's been set previously to degrees.  This also seems to be an issue with the get commands of amp_vpp and amp_vrms (haven't tried amp_dbm).  My AFG is being used in an ongoing measurement right now, but I wonder if just added the units RAD/DEG like so:

'phase:adjust RAD?'
'phase:adjust DEG?' 

would give the query back in the desired units.

Comment (FlyingCurryMonster) [reply pymeasure/instruments/tektronix/afg3152c.py]: also apologies for the late replies, I've been working on my final required graduate course for my degree program.

Comment (bilderbuchi) [reply pymeasure/instruments/tektronix/afg3152c.py]: Good to have this confirmed. Such statefulness in instruments/command sets is always annoying.
To answer your question, I guess you need to study the manual and/or try it out when the instrument is free again.

**Comment by BenediktBurger** [comment]
Hey @FlyingCurryMonster , do you want to pick this PR up again or not?

**Comment by FlyingCurryMonster** [comment]
For sure, apologies I've been busy with apparatus development.

**Pull request closed without merging**